src/utility/AddPublicationUtility.js

```diff
@@ -0,0 +1,254 @@
+  let updatedInput = { ...input };
+
+  // update publications, phenotypes, variant_descriptions, mechanism_evidence fields of input
+  publicationsData.results.forEach((item) => {
+    updatedInput.publications[item.pmid] = {
+      families: null,
+      affectedIndividuals: null,
+      consanguineous: "unknown",
+      ancestries: "",
+      comment: "",
+      source: item.source,
+      year: item.year,
+      title: item.title,
+      authors: item.authors,
+    };
+
+    updatedInput.phenotypes[item.pmid] = {
+      summary: "",
+      hpo_terms: [],
+    };
+
+    updatedInput.variant_descriptions[item.pmid] = {
+      description: "",
+    };
+
+    let evidenceTypesObj = {};
+    EvidenceTypesAttribs.forEach((item) => {
+      evidenceTypesObj[item.primaryType] = [];
+    });
+    updatedInput.mechanism_evidence[item.pmid] = {
+      description: "",
+      evidence_types: evidenceTypesObj,
+    };
+  });
+
+  return updatedInput;
+};
+
+export const updateInputWithRemovedPublications = (input, removedPmidList) => {
+  let updatedInput = { ...input };
+
+  removedPmidList.forEach((pmid) => {
+    // remove pmid from publications, phenotypes, variant_descriptions, mechanism_evidence fields of input
+    delete updatedInput.publications[pmid];
+    delete updatedInput.phenotypes[pmid];
+    delete updatedInput.variant_descriptions[pmid];
+    delete updatedInput.mechanism_evidence[pmid];
+  });
+
+  // remove pmid from supporting papers list of variant_types
+  for (let primaryTypeKey in updatedInput.variant_types) {
+    for (let secondaryTypeKey in updatedInput.variant_types[primaryTypeKey]) {
+      const supportingPapers =
+        updatedInput.variant_types[primaryTypeKey][secondaryTypeKey]
+          .supporting_papers;
+      // filter supporting papers by removing pmids
+      const filteredSupportingPapers = supportingPapers.filter(
+        (supportingPaper) => !removedPmidList.includes(supportingPaper)
+      );
+      updatedInput.variant_types[primaryTypeKey][
+        secondaryTypeKey
+      ].supporting_papers = filteredSupportingPapers;
+    }
+  }
+
+  return updatedInput;
+};
+
+export const updateHpoTermsInputHelperWithNewPublicationsData = (
+  hpoTermsInputHelper,
+  pmidList
+) => {
+  let updatedHpoTermsInputHelper = { ...hpoTermsInputHelper };
+  pmidList.forEach((pmid) => {
+    updatedHpoTermsInputHelper[pmid] = {
+      isHpoTermsDataLoading: false,
+      hpoTermsErrorMsg: "",
+      isHpoTermsValid: true,
+      hpoTermsInput: "",
+    };
+  });
+  return updatedHpoTermsInputHelper;
+};
+
+export const updateHpoTermsInputHelperWithRemovedPublications = (
+  hpoTermsInputHelper,
+  removedPmidList
+) => {
+  let updatedHpoTermsInputHelper = { ...hpoTermsInputHelper };
+  removedPmidList.forEach((pmid) => {
+    delete updatedHpoTermsInputHelper[pmid];
+  });
+  return updatedHpoTermsInputHelper;
+};
+
+export const prepareInputForNewPublicationDataSubmission = (input) => {
```

Comment:
This should only includes data that is going to be added/updated.

Reply:
Addressed in latest commit